data-node/src/main/java/org/graylog/datanode/Configuration.java

```diff
@@ -246,22 +248,24 @@ public String getOpensearchConfigLocation() {
      * This is a pointer to a directory holding configuration files (and certificates) for the datanode itself.
      * We treat it as read only for the datanode and should never persist anything in it.
      * Use {@link DatanodeDirectories} to obtain a reference to this directory.
+     *
      */
-    public String getDatanodeConfigurationLocation() {
+    @Nullable
```

Comment:
Is @Nullable really working here?

Reply:
What do you mean by "working"? I added the annotation just to signalize that the returned value can be null.

Reply:
I've never seen this annotation to be used for return values. but I'm fine with it.

Reply:
Ah, now I understand. It's indeed working: 

![image](https://github.com/Graylog2/graylog2-server/assets/4102775/c59aa803-b75e-4d45-8247-629b50bfe912)